runtime/src/chpl-launcher-common.c

```diff
@@ -102,6 +103,42 @@ void chpl_append_to_largv(int* largc, const char*** largv, int* largv_len,
   (*largv)[(*largc)++] = (arg);
 }
 
+// Helper for appending arguments to a variable-size command buffer.
+// - Requires the buffer pointer is uninitialized on first call.
+// - After exceeding an initial allocated size estimate, each call will allocate
+//   additional memory as needed.
+char* chpl_append_to_cmd(char* cmdBuf, const char* format, ...) {
```

Comment:
as an API concern, I'm kinda concerned that `charsWritten` is a local static variable, and in theory `chpl_append_to_cmd` could be called with multiple buffers. Consider either making it explicit that `chpl_append_to_cmd` should only ever be called with 1 buffer, or requiring users of it to track the charsWritten as a passed in `int*` as well